pgrx/src/bgworkers.rs

```diff
@@ -121,6 +125,24 @@ impl BackgroundWorker {
         GOT_SIGTERM.swap(false, Ordering::SeqCst)
     }
 
+    /// Have we received a SIGINT?
+    pub fn sigint_received() -> bool {
+        unsafe {
+            assert!(!pg_sys::MyBgworkerEntry.is_null(), "BackgroundWorker associated functions can only be called from a registered background worker");
+        }
+        // toggle the bool to false, returning whatever it was
+        GOT_SIGINT.swap(false, Ordering::SeqCst)
+    }
+
+    /// Have we received a SIGCHLD?
+    pub fn sigchld_received() -> bool {
```

Comment:
Please add documentation to these functions that indicates they mutate the worker's state, so you **better** actually use this boolean instead of just calling it for funsies. Perhaps also with something like `#[must_use = "you aren't getting the same bool back if you call this twice"]`?

You may update the other functions already like these two for consistency, or allow that to happen in another PR, but I do not want to merge another one of these without such.

Reply:
done